index.html

```diff
@@ -2175,6 +2231,130 @@ <h3>Verify Proof Sets and Chains</h3>
         </ol>
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+The purpose of the algorithm in this section is to ensure that a consuming
+application has explicitly approved of the types, and therefore the semantics,
+of input documents that it will process. Not checking JSON-LD context values
+against known good values can lead to security vulnerabilities, due to variance
+in the semantics that they convey. The context validation algorithm is one
+that applications validating a [=conforming secured document=] MUST run after
+running the algorithm in Section [[[#verify-proof]]] or Section
+[[[#verify-proof-sets-and-chains]]]. This algorithm takes a document ([=map=]
+|inputDocument|), a set of approved JSON-LD Contexts ([=map=]
+|approvedContexts|), and a boolean to recompact when unknown contexts are
+detected ([=boolean=] |recompact|) and returns a [=map=] that contains the
+following:
+        </p>
+
+        <ul>
+          <li>
+a status ([=boolean=] |status|)
+          </li>
+          <li>
+a validated document ([=map=] |document|) if no errors occurred
+          </li>
+          <li>
+zero or more warnings ([=list=] of [=ProblemDetails=] |warnings|)
+          </li>
```

Comment:
This isn't used in the algorithm.

```suggestion
```

Reply:
Yes, but it's left in there so that implementations can add warnings if they would like (and there is a standards interface for them doing so. I'll add some text to note that implementations MAY add additional warnings/errors.

Reply:
Fixed in 04584e7b095ab6afbbaa83a6221f01db5f64ef2c.